test/python/synthesis/test_sqisw_synthesis.py

```diff
@@ -0,0 +1,68 @@
+# This code is part of Qiskit.
+#
+# (C) Copyright IBM 2017, 2019.
+#
+# This code is licensed under the Apache License, Version 2.0. You may
+# obtain a copy of this license in the LICENSE.txt file in the root directory
+# of this source tree or at http://www.apache.org/licenses/LICENSE-2.0.
+#
+# Any modifications or derivative works of this code must retain this
+# copyright notice, and modified files need to carry a notice indicating
+# that they have been altered from the originals.
+
+# pylint: disable=invalid-name
+
+"""Test synthesis of two-qubit unitaries into SQiSW gates."""
+
+import unittest
+from test import combine
+from ddt import ddt
+
+from qiskit.synthesis.two_qubit import SQiSWDecomposer
+from qiskit.quantum_info import random_unitary, Operator
+from qiskit.circuit.library import SwapGate, iSwapGate, CXGate, IGate
+from qiskit.test import QiskitTestCase
+
+
+@ddt
+class TestSQiSWSynth(QiskitTestCase):
```

Comment:
It would be nice to add a test for each of the different regions (green, blue, brown and red) in the paper so we're sure each of the cases is triggered 🙂

Reply:
i added a bunch of tests